framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3094,15 +3095,35 @@ protected void checkThrownExpression(ThrowTree tree) {
         AnnotatedTypeMirror throwType = atypeFactory.getAnnotatedType(tree.getExpression());
         TypeMirror throwTM = throwType.getUnderlyingType();
         Set<? extends AnnotationMirror> required = getThrowUpperBoundAnnotations();
+        if (methodTree != null && getExceptionList(methodTree) != null) {
+            List<AnnotatedTypeMirror> exceptionList = getExceptionList(methodTree);
+            for (AnnotatedTypeMirror exception : exceptionList) {
+                Types typesUtil = atypeFactory.getProcessingEnv().getTypeUtils();
+                if (typesUtil.isSubtype(
+                        exception.getUnderlyingType(), throwType.getUnderlyingType())) {
+                    required = exception.getEffectiveAnnotations();
+                    break;
+                }
+            }
+        }
```

Comment:
Should there be a warning or error if no specific required type is found?
When should this use `getThrowUpperBoundAnnotations();` ?
Could you make that initialization more explicit, to make clear it is intentional?

Reply:
> Should there be a warning or error if no specific required type is found? When should this use `getThrowUpperBoundAnnotations();` ? Could you make that initialization more explicit, to make clear it is intentional?

I am think about because there is not always a throw clause in the method signature.  Consider the following example. If there is not match, there should not be warning or error.

```
void foo() {
    try {
         // code might throw an exception
    } catch (SomeCheckedException e) {

    }
}
```

Reply:
You didn't come back to this comment. Doesn't this mean that from a `throw` statement, you need to look at both the enclosing `catch` clauses and only finally at the `throws` clause on the enclosing method?

Reply:
Do you mean code like this shoule issue error?
```
void foo() {
    try {
        throw new @Initalized SomeCheckedException();
    } catch (@UnderInitialization SomeCheckedException e) {

    }
}
```

Reply:
I think in that case we would need a `visitry` method in basetypevisitor to get the try block and compare it with possible multiple catch block.

Reply:
I am not sure why it is not shown in the first page of this PR directly but I do reply the comment here yesterday.